src/lib/storage/storage_manager.cpp

```diff
@@ -10,38 +12,44 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager _storage_manager;
+
+  return _storage_manager;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
+  _tables.insert(std::make_pair(name, table));
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
+  DebugAssert(has_table(name), "The storage manager doesn't know the requested table");
+  _tables.erase(name);
 }
 
-std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  // Implementation goes here
-  return nullptr;
-}
+std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const { return _tables.at(name); }
 
-bool StorageManager::has_table(const std::string& name) const {
-  // Implementation goes here
-  return false;
-}
+bool StorageManager::has_table(const std::string& name) const { return _tables.count(name) != 0; }
 
 std::vector<std::string> StorageManager::table_names() const {
-  throw std::runtime_error("Implement StorageManager::table_names");
+  std::vector<std::string> table_names;
+  for (auto const& table : _tables) {
+    table_names.push_back(table.first);
+  }
+
+  std::sort(table_names.begin(), table_names.end());
```

Comment:
Was that a requirement?

Reply:
No it wasn't but the implementation of `unordered_map` specifically states there are 

> no guarantees which specific element is considered its first element.
So I thought this would make the function more predictable and the number of `table_names` should be relatively small.